src/main/java/org/isf/menu/manager/UserBrowsingManager.java

```diff
@@ -334,37 +319,49 @@ public UserGroup newUserGroup(UserGroup userGroup, List<Permission> permissions)
 		String code = userGroup.getCode();
 		if (ioOperations.isGroupNamePresent(code)) {
 			throw new OHDataIntegrityViolationException(
-					new OHExceptionMessage(MessageBundle.formatMessage("angal.groupsbrowser.thegroupalreadyexists.fmt.msg", code)));
+				new OHExceptionMessage(MessageBundle.formatMessage("angal.groupsbrowser.thegroupalreadyexists.fmt.msg", code)));
 		}
 		return ioOperations.newUserGroup(userGroup, permissions);
 	}
 
 	/**
 	 * Updates an existing {@link UserGroup} in the DB.
-	 *
 	 * @param userGroup - the {@link UserGroup} to update
-	 * @return {@code true} if the group has been updated, {@code false} otherwise.
+	 * @return the {@link UserGroup} that has been updated
+	 * @throws OHServiceException If failed to update group
 	 */
-	public boolean updateUserGroup(UserGroup userGroup) throws OHServiceException {
+	public UserGroup updateUserGroup(UserGroup userGroup) throws OHServiceException {
+
+		UserGroup group = ioOperations.findByCodeAndIsDeleted(userGroup.getCode());
+
+		if (group != null && group.isDeleted() && userGroup.isDeleted()) {
```

Comment:
Isn't `findByCodeAndIsDeleted` already returning a isDeleted group (or `null`)?
If yes, the condition `... && group.isDeleted() ...` is for sure `true`?